src/authentication-strategies/JWT.strategy.ts

```diff
@@ -0,0 +1,26 @@
+const jwt = require('jsonwebtoken');
+import {promisify} from 'util';
+const signAsync = promisify(jwt.sign);
+const verifyAsync = promisify(jwt.verify);
+// Consider turn it to a binding
+const SECRET = 'secretforjwt';
+import {Request} from '@loopback/rest';
+
+export class JWTStrategy {
+  // tslint:disable-next-line:no-any
+  async authenticate(request: Request): Promise<any> {
+    // there is a discussion regarding how to retrieve the token,
+    // see comment https://github.com/strongloop/loopback-next/issues/1997#issuecomment-451054806
+    const token = request.query.token || request.headers['authorization'];
+
+    if (token) {
+      try {
+        return await verifyAsync(token, SECRET);
+      } catch (err) {
+        if (err) return Promise.reject('Authentication failed!');
+      }
+    }
+
+    return Promise.reject('Token not found!');
```

Comment:
When the client did not provide any token, perhaps because it did not know this URL requires authentication, the server should return `401 Unauthorized` response with `WWW-Authenticate` header field.

In Passport, the strategy would call `this.fail()` API, see e.g. how `passport-jwt` uses it here: https://github.com/themikenicholson/passport-jwt/blob/f2044371bc9c85f22fbd30e8e58267a42dcd26ac/lib/strategy.js#L96

In `@loopback/authentication`, the strategy adapter is implementing `fails` as follows. I think this is not entirely correct implementation, e.g. I don't see any `WWW-Authenticate` header field being set.

https://github.com/strongloop/loopback-next/blob/aacfcf7b1ee8f6550b91d1fa3c84a6e4fbfc9a36/packages/authentication/src/strategy-adapter.ts#L50-L53
```ts
      // add failure state handler to strategy instance
      strategy.fail = function(challenge: string) {
        reject(new HttpErrors.Unauthorized(challenge));
      };
```